app/src/main/java/com/github/leuludyha/ibdb/presentation/components/auth/signup/add_friends/AddFriendsFromContactsPrompt.kt

```diff
@@ -0,0 +1,213 @@
+        private val getUserFromPhoneNumberUseCase: GetUserFromPhoneNumberUseCase
+    ) : ViewModel() {
+
+        /**
+         * Find all contacts of the user, and map them to either a phone number or an email
+         * "Copy-pasted" from https://stackoverflow.com/questions/12562151/android-get-all-contacts
+         */
+        @SuppressLint("Range")
+        fun getContacts(ctx: Context): List<Contact> {
+            val result = mutableListOf<Contact>()
+            val cr: ContentResolver = ctx.contentResolver
+            val cur: Cursor? = cr.query(
+                ContactsContract.Contacts.CONTENT_URI,
+                null, null, null, null
+            )
+            if ((cur?.count ?: 0) > 0) {
+                while (cur != null && cur.moveToNext()) {
+                    val id: String = cur.getString(
+                        cur.getColumnIndex(ContactsContract.Contacts._ID)
+                    )
+                    val name: String = cur.getString(
+                        cur.getColumnIndex(
+                            ContactsContract.Contacts.DISPLAY_NAME
+                        )
+                    )
+                    if (cur.getInt(
+                            cur.getColumnIndex(
+                                ContactsContract.Contacts.HAS_PHONE_NUMBER
+                            )
+                        ) > 0
+                    ) {
+                        val pCur: Cursor? = cr.query(
+                            ContactsContract.CommonDataKinds.Phone.CONTENT_URI,
+                            null,
+                            ContactsContract.CommonDataKinds.Phone.CONTACT_ID + " = ?",
+                            arrayOf(id),
+                            null
+                        )
+                        if (pCur != null) {
+                            while (pCur.moveToNext()) {
+                                val phoneNo: String = pCur.getString(
+                                    pCur.getColumnIndex(
+                                        ContactsContract.CommonDataKinds.Phone.NUMBER
+                                    )
+                                )
+
+                                result.add(
+                                    Contact(
+                                        name = name,
+                                        phoneNumber = phoneNo,
+                                        email = null
+                                    )
+                                )
+                            }
+                            pCur.close()
+                        }
+                    }
+                }
+            } else {
+                Log.e("FRIENDS", "No Contacts !")
+            }
+            cur?.close()
+
+            return result
+        }
+
+        /**
+         * Return the list of existing users in the database which this user make know
+         */
+        fun getPossibleAcquaintances(contacts: List<Contact>): List<User> {
```

Comment:
This method doesn't work for now because of `Firebase`... We can merge it for now but it needs to be fixed quickly so that we can demo on Friday.